src/server/scripts/Pet/pet_hunter.cpp

```diff
@@ -95,22 +95,22 @@ class npc_pet_hunter_snake_trap : public CreatureScript
                     Unit* summoner = me->ToTempSummon()->GetSummonerUnit();
 
                     std::vector<Unit*> targets;
-                    for (std::pair<ObjectGuid const, PvPCombatReference*> const& pair : summoner->GetCombatManager().GetPvPCombatRefs())
+
+                    auto check = [&](auto const& pair) mutable
```

Comment:
this is really unintuitive, you be better if you passed the entire container as argument, instead of its element
and named it addValidTargets

oh and dont auto the arg, its `std::pair<ObjectGuid const, CombatReference*> const&` (auto ::iterators, not pairs)

Reply:
> oh and dont auto the arg, its std::pair<ObjectGuid const, CombatReference*> const& (auto ::iterators, not pairs)

I don't understand what exactly you mean?

Reply:
We only use auto for iterators (.begin, .end) or when its required (for example lambda variable)

It is not required on the argument (pair) so write out the type, dont use `auto`

Reply:
You mean the `pair` parameter in lambda expression?